onnx_chainer/functions/array.py

```diff
@@ -316,3 +317,44 @@ def convert_Where(func, opset_version, input_names, num_outputs, context,
                   parameters):
     input_names.insert(0, context.get_name(func.condition))
     return onnx_helper.make_node('Where', input_names, num_outputs),
+
+
+# NOTE(syoyo): `Upsampling` is deprecated in ONNX opset 10.
+# Use `Reshape` for opset 10
+@support((7, 9))
+def convert_ResizeImages(func, opset_version, input_names, num_outputs,
+                        context, parameters):
+
+    warnings.warn(
+        '`resize_images` is mapped to `Upsampling` ONNX op with bilinear '
+        'interpolation. '
+        'Behavior of bilinear interpolation differs from each implementation. '
+        'See the issue https://github.com/chainer/onnx-chainer/issues/147 '
+        'for details.',
+        UserWarning)
+
+    outsize = (func.out_H, func.out_W)
+
+    h, w = func.inputs[0].shape[2:]
+
+    # Compute scaling factor.
+    # NOTE(syoyo): Despite of its name, `Upsample` onnx op will downsample
+    # images when scale value is less than 1.0
+    scales = [1.0, 1.0, float(outsize[0]) / float(h), float(outsize[1]) / float(w)]
+
+    if (scales[2] < 1.0e-8) and (scales[3] < 1.0e-8):
```

Comment:
scales = `[1.0, 10, 2.0, 1.0e-9]` seems acceptable, is `and` OK?

Reply:
I'm not sure how Chainer and ONNX(Runtime) behaves when scaling value is near to zero. 

If (width=0.0, height=128) will create (valid) 1x128 Tensor, we should accept such scaling parameter input(i.e. must use `and` for checking scale values). If Chainer and ONNX accepts scaling parameter 0.0 both for width and height, we can omit this scaling value check.

Reply:
OK, I also don't know how the frameworks deal with such a small value, but anyway, can understand the policy, thank you.